File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/DoctorSpecialities/DoctorSpecialitiesForm.cs
using Serenity.ComponentModel;

namespace HosbitalSerenity.Hosbital.Forms;

[FormScript("Hosbital.DoctorSpecialities")]
[BasedOnRow(typeof(DoctorSpecialitiesRow), CheckNames = true)]
public class DoctorSpecialitiesForm
{
    public int DoctorId { get; set; }
    public int SpecialityId { get; set; }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/DoctorSpecialitiesForm.ts
import { ServiceLookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";

export interface DoctorSpecialitiesForm {
    DoctorId: ServiceLookupEditor;
    SpecialityId: ServiceLookupEditor;
}

export class DoctorSpecialitiesForm extends PrefixedContext {
    static readonly formKey = 'Hosbital.DoctorSpecialities';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!DoctorSpecialitiesForm.init)  {
            DoctorSpecialitiesForm.init = true;

            var w0 = ServiceLookupEditor;

            initFormType(DoctorSpecialitiesForm, [
                'DoctorId', w0,
                'SpecialityId', w0
            ]);
        }
    }
}